shell/platform/android/io/flutter/plugin/common/MethodCall.java

```diff
@@ -0,0 +1,29 @@
+package io.flutter.plugin.common;
+
+import java.util.Objects;
+
+/**
+ * Command object representing a method call on a {@link FlutterChannel}.
+ */
+public final class MethodCall {
+    /**
+     * The name of the called method.
+     */
+    public final String method;
+
+    /**
+     * Arguments for the call.
+     */
+    public final Object arguments;
```

Comment:
I've separated messages from method calls and event streams, so that we now have two codec concepts (`MessageCodec<T>`, `MethodCodec`) and two channel concepts (`FlutterMessageChannel<T>`, `FlutterMethodChannel`). Event streams are implemented using the `MethodCodec`, and I have not (yet) created a separate channel type for that. PTAL and let me know what you think.